### DISCUSSION TITLE
Advanced Deposit/Restock Work with Slot Memory.

### DISCUSSION THREAD

**Discussion by HeavyDeception** [P3pp3rF1y/SophisticatedBackpacks#482] [Feature / Idea Archive]
This is kind of for both Storage and Backpacks.
When giving slots Memory of and item Deposit/Restock will put those items in those slots when you have it set to Match Inventory Contents.

**Comment by MagisterXero** [comment]
I'd like to add on to this by making a request for the Deposit upgrade specifically.
For example, I'd prefer the deposit upgrade, in addition to the filter, also NOT deposit items placed within a designated row (IE 'no sort' or something equivalent)

Great mod btw! Absolutely love using it, it's my goto for backpacks.

**Comment by antisocialian** [comment]
yes, i usually set my tools/books to the bottom rows and then the mining stuff goes in the tops ones. then when i quick deposit i want to keep my tools. unfortunately the filters for sophisticated storage don't limit the quick deposit and my tools will still go into that barrel. having to setup a drawer system just for raw ore chunks or ore blocks seems excessive as a work around.

maybe even a filter item that lets me to a larger list of items, or something similar to the memory where i can lock specific slots to never deposit from(or the inverse where only certain spots get deposited from)

**Comment by BigJMoney** [comment]
I can agree with all of this, but I mostly want what is mentioned in the 2nd comment: I would like for the deposit upgrade (or advanced deposit upgrade) to ignore item memory slots. This can be the default behavior, because there is never a case where I want memory slots to be deposited (if I did, they would not be memory slots!). But if you want to make this an option that can be toggled, of course that would also be great.

I can try to use the item filter for this behavior, but there are two issues doing that:

1. Now I can't use the 'match inventory' option, which is mutually exclusive from the item filter
2. I don't want to completely filter out an item type. I only want the memory slot itself to be ignored. For example, I have a stack of iron in a memory slot so that I always have 1 stack on me. Then I have 5 more stacks of iron. I don't want to prevent all of the iron from being deposited.

TO CLARIFY: It occurred to me that it might sound like I'm asking for the opposite of what the OP is asking for, but this is not the case. I'm only referring to the situation where the backpack is using memory slots (ignore the slots). I think the OP is referring to the case where the storage container has the memory slots (utilize the memory slots). So to be extra clear, when a backpack has the deposit upgrade, it should NEVER deposit FROM memory slots on the backpack, but should ALWAYS deposit TO memory slots on the storage container (even if they are empty, when using the match inventory option).

**Comment by MasterCakeX5** [comment]
Please please make this  a feature you can configure! It really does not make any sense to have the deposit to chest option drawing from memory or not having a configure option to do this. I have a stack of redstone or logs for example, I want to deposit everything BUT that stack - and that stack sits in a memory slot.
